ext/ddtrace_profiling_native_extension/collectors_thread_context.c

```diff
@@ -777,8 +779,11 @@ static void initialize_context(VALUE thread, struct per_thread_context *thread_c
       StringValueCStr(invoke_file_location),
       invoke_line_location
     );
-  } else {
-    snprintf(thread_context->thread_invoke_location, THREAD_INVOKE_LOCATION_LIMIT_CHARS, "%s", "");
+  } else if (thread != state->main_thread) {
+    // If the first function of a thread is native code, there won't be an invoke location, so we use this fallback.
+    // NOTE: In the future, I wonder if we could take the pointer to the native function, and try to see if there's a native
+    // symbol attached to it.
+    snprintf(thread_context->thread_invoke_location, THREAD_INVOKE_LOCATION_LIMIT_CHARS, "%s", "(Unnamed thread from native gem)");
```

Comment:
Is it always a thread from a "gem"? Could this thread have originated from the current application's native code?

Reply:
You know what, your question gave me quite a bit of pause.

This being Ruby, the answer is of course "everything is possible", but then I started thinking it would be kinda awkward it would be, since you wouldn't get the automatic compilation behavior on install like you get for gems. On the other hand, I can imagine if you need to interact with some weird hardware (like for a point of sale), you may end up doing that anyway.

In the end, as we already had another placeholder "In native code", I changed this one to be `(Unnamed thread from native code)` to match it in  c52c7329079e801dffac1256dd887ad283610ccd .